Feature/ysmod 328 saksbehandling db modell

lurajon opened the pull request.

lurajon committed:
YSMOD-328

- Opprettet JPA entiteter for skademelding
- Opprettet repository og service for skademelding
- Opprettet graphql modell og tjeneste
- Opprettet tester for graphql og service

lurajon committed:
YSMOD-328

- fjernet skademelding db modell
- la til db modell for sak, behandling, dokument, dokumentfil
- opprettet kafka consumer for oppgave-opprettet topic
- tester

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/config/JacksonConfig.kt:
```diff
@@ -0,0 +1,22 @@
+    fun objectMapper(builder: Jackson2ObjectMapperBuilder): ObjectMapper {
+        val objectMapper: ObjectMapper = builder.build()
+        objectMapper.registerModule(JavaTimeModule())
+       // objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
```
Fjernes?

lurajon replied:
Fjernes!

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/hendelser/oppgave/model/Ident.kt:
```diff
@@ -0,0 +1,6 @@
+package no.nav.yrkesskade.saksbehandling.hendelser.oppgave.model
+
+class Ident (
+    val verdi: String,
```
Hva er Ident og hva er verdi? Kan de få bedre navn?

lurajon replied:
Ident er kopiert fra Oppgave - Ident har både fnr og aktør id.

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/hendelser/oppgave/model/OppgaveRecord.kt:
```diff
@@ -0,0 +1,28 @@
+    val oppgavetype: String,
+    val fristFerdigstillelse: LocalDate,
+    val aktivDato: LocalDate,
+    val opprettetTidspunkt: LocalDateTime,
```
Rydde litt i rekkefølgen på feltene, slik at alt som har med opprettet/endret/status kommer til sist. Hadde kanskje vært penest å trekke disse ut i en egen klasse?

lurajon replied:
Bra forslag!

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/hendelser/oppgave/model/OppgaveRecord.kt:
```diff
@@ -0,0 +1,28 @@
+    val journalpostId: String?,
+    val temagruppe: String?,
+    val tema: String,
+    val behandlingstema: String,
```
er tema og behandlingstema to forskjellige ting?

lurajon replied:
Det ser ut som det er to forskjellige ting i oppgave modellen ja. tema er YRK, mens behandlingstema er en av disse: https://kodeverk-web.dev.intern.nav.no/kodeverksoversikt/kodeverk/Behandlingstema

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/hendelser/oppgave/model/Oppgavestatus.kt:
```diff
@@ -0,0 +1,37 @@
+        }
+
+        fun getIdsFor(statuskategori: Oppgavestatuskategori?): List<Long?> {
+            return if (Objects.equals(Sakstatus.AAPEN, statuskategori)) {
```
Sammenligning av Saksstatus og Oppgavestatuskategori?

lurajon replied:
Bra sett! Fikses

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/BehandlingEntity.kt:
```diff
@@ -0,0 +1,64 @@
+    val behandlingId: Long,
+
+    @Column(name = "behandlingsansvarlig_ident", nullable = true)
+    val behandlingsAnsvarligIdent: String?,
```
litt flisespikking; behandlingsansvarligIdent

lurajon replied:
Fikset

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/BehandlingEntity.kt:
```diff
@@ -0,0 +1,64 @@
+
+    @Enumerated
+    @Column(name = "status", nullable = false)
+    val status: Behandlingsstatus,
```
--endre kolonnenavn til behandlingsstatus eller type til Status?--
Jeg tar det tilbake :)

lurajon replied:
ok :)

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/BehandlingResultatEntity.kt:
```diff
@@ -0,0 +1,29 @@
+
+@Entity
+@Table(name = "behandling_resultat")
+class BehandlingResultatEntity(
```
Er BehandlingsresultatEntity et bedre navn?

lurajon committed:
YSMOD-328

- la til scalar for Instant

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/BehandlingResultatEntity.kt:
```diff
@@ -0,0 +1,29 @@
+    val sted: String,
+
+    @Column(name = "bakgrunn_aarsak")
+    val bakgrunnAarsak: String,
```
bakgrunnsaarsak

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/BehandlingResultatEntity.kt:
```diff
@@ -0,0 +1,29 @@
+    @Id
+    @Column(name = "behandling_resultat_id", nullable = false, updatable = false)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    val behandlingResultatId: Long,
```
og her

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/DokumentFilEntity.kt:
```diff
@@ -0,0 +1,29 @@
+
+@Entity
+@Table(name = "dokument_fil")
+class DokumentFilEntity (
```
DokumentfilEntity?

lurajon committed:
YSMOD-328

- la til scalar for Instant

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/DokumentFilEntity.kt:
```diff
@@ -0,0 +1,29 @@
+
+    @ManyToOne
+    @JoinColumn(name = "dokument_id")
+    val dokument: DokumentMetaEntity
```
dokumentMeta? Men det et klinger ikke særlig godt...

lurajon replied:
Enig! Egentlig er alle tabellene våre meta fra dokument og oppgave. Jeg endrer til kun Dokument

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/DokumentMetaEntity.kt:
```diff
@@ -0,0 +1,43 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    val dokumentId: Long,
+
+    @Column(name = "journal_post_id", nullable = false)
```
kolonnenavn journalpost_id

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/Dokumentkategori.kt:
```diff
@@ -0,0 +1,10 @@
+package no.nav.yrkesskade.saksbehandling.model
+
+enum class Dokumentkategori {
+    MEDISINKE_OPPLYSNINGER,
```
typo; MEDISINSKE_

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/Dokumentkategori.kt:
```diff
@@ -0,0 +1,10 @@
+    ARBEIDSAVTALE_HJEMMEKONTOR,
+    SPESIALIST,
+    SKADEFORKLARING,
+    TILLEGGSKJEMA,
```
TILLEGGSSKJEMA

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/model/SakEntity.kt:
```diff
@@ -0,0 +1,45 @@
+
+    @Enumerated
+    @Column(name = "sak_status", nullable = false)
+    val sakstatus: Sakstatus,
```
saksstatus: Saksstatus

erlendkri commented on src/main/resources/db/migration/V2022_05_25__YSMOD-328-saksbehandling.sql:
```diff
@@ -0,0 +1,79 @@
+    sakstype             VARCHAR(50)  NOT NULL,
+    aktoer_id            VARCHAR(20)  NOT NULL, -- brukes av flere systemer i NAV
+    bruker_identifikator VARCHAR(12)  NOT NULL, -- fnr/dnr
+    bruker_fornavn       VARCHAR(100) NOT NULL,
```
bør alle brukerfeltene ligge rett på sak?

lurajon replied:
Ja. Det er by design for å kunne iverata historiske data for navn og id

erlendkri commented on src/main/resources/db/migration/XX_V2022_05_16__YSMOD-328-skademelding-modell.sql:
```diff
@@ -0,0 +1,93 @@
+  -- innmelder
+    innmelder_identitetsnummer VARCHAR(15) NOT NULL,
+  	innmelder_paa_vegne_av VARCHAR (15) NOT NULL,
+  	innmelder_virksomhetsnavn VARCHAR (255) NOT NULL DEFAULT '',
```
er det bedre å tillate null enn å sette default tom streng?

erlendkri commented on src/main/resources/db/migration/XX_V2022_05_16__YSMOD-328-skademelding-modell.sql:
```diff
@@ -0,0 +1,93 @@
+  	skadelidt_rolletype VARCHAR (100) NOT NULL,
+  -- dekningsforhold
+  	dekning_organisasjonsnummer VARCHAR (15) NOT NULL,
+  	dekning_navn_paa_virksomheten VARCHAR (255) NOT NULL DEFAULT '',
```
samme her

lurajon replied:
Denne modell filen slettes

erlendkri commented on src/main/resources/db/migration/XX_V2022_05_16__YSMOD-328-skademelding-modell.sql:
```diff
@@ -0,0 +1,93 @@
+	hendelse_naar_skjedde_ulykken VARCHAR (100) NOT NULL,
+  	hendelse_hvor_skjedde_ulykken VARCHAR (100) NOT NULL,
+  	hendelse_utfyllende_beskrivelse VARCHAR (2000) NOT NULL,
+  	hendelse_steds_beskrivelse VARCHAR (100) NOT NULL,
```
_stedsbeskrivelse evt _sted_beskrivelse

erlendkri reviewed (commented):
Imponerende mengde arbeid! 
Jeg har prøvd å være pirkete på navngiving i datamodellen. Du får vurdere om du vil endre...

lurajon committed:
YSMOD-328

- fikset diverse kodegjennomgang endringer
- flyttet setting av payload i hendelse til før kall metoden
- endret id navn på kafkalytteren

lurajon committed:
YSMOD-328

- la til db sjekk på hendelse test

lurajon committed:
YSMOD-328

- fikset merge conflict
- skrudde av autostart av oppgave hendelser og bruker unleash kill switch til å definere om den skal startes eller ikke - alltid på i lokal og test

lurajon committed:
YSMOD-328

- fikset secret navn

lurajon committed:
YSMOD-328

- disable CORS på graphql

lurajon committed:
YSMOD-328

- wildcard cors
